File: src/main/kotlin/tables/Students.kt
package tables

import org.jetbrains.exposed.v1.core.Table
import org.jetbrains.exposed.v1.datetime.date
import org.jetbrains.exposed.v1.datetime.datetime
import schema.type.Gender
import schema.type.CountryCode

object Students : Table() {
    val id = integer("id").autoIncrement()
    val name = varchar("name", 255)
    val email = varchar("email", 255).nullable()
    val phoneNumber = varchar("phone_number", 255).nullable()
    val dateOfBirth = date("date_of_birth").nullable()
    val gender = enumerationByName("gender", 10, Gender::class).nullable()
    val nationality = enumerationByName("nationality", 2, CountryCode::class).nullable()
    val deletedAt = datetime("deleted_at").nullable()
    val createdAt = datetime("created_at")
    val updatedAt = datetime("updated_at")

    override val primaryKey = PrimaryKey(id)
}

File: db/schema/students.ts
import {
    pgTable,
    serial,
    varchar,
    timestamp,
    pgEnum,
} from "drizzle-orm/pg-core";
import { relations } from "drizzle-orm";
import { certificates, templateRecipientGroupItems } from ".";

export const genderEnum = pgEnum("Gender", ["MALE", "FEMALE", "OTHER"]);
export const countryCodeEnum = pgEnum("CountryCode", ["EG", "US"]);

export const students = pgTable("Student", {
    id: serial("id").primaryKey(),
    name: varchar("name", { length: 255 }).notNull(),
    email: varchar("email", { length: 255 }),
    phoneNumber: varchar("phoneNumber", { length: 255 }),
    dateOfBirth: timestamp("dateOfBirth", { precision: 3 }),
    gender: genderEnum("gender"),
    nationality: countryCodeEnum("nationality"),
    createdAt: timestamp("createdAt", { precision: 3 }).notNull(),
    updatedAt: timestamp("updatedAt", { precision: 3 }).notNull(),
});

export const studentsRelations = relations(students, ({ many }) => ({
    recipientGroupItems: many(templateRecipientGroupItems),
    certificates: many(certificates),
}));
